aioresponses/compat.py

```diff
@@ -16,6 +16,9 @@
 
 AIOHTTP_VERSION = parse_version(aiohttp_version)
 
+stream_reader_factory: Callable[
+    [Optional[asyncio.AbstractEventLoop]], StreamReader
+]
 if AIOHTTP_VERSION >= parse_version('3.0.0'):
```

Comment:
If mypy reports that `error: All conditional function variants must have identical signatures [misc]`, wouldn't it be better to just add correct type annotations to both variants of `stream_reader_factory`?
```py
if AIOHTTP_VERSION >= parse_version('3.0.0'):
    from aiohttp.client_proto import ResponseHandler

    def stream_reader_factory(
        loop: Optional[asyncio.AbstractEventLoop] = None
    ) -> StreamReader:
        protocol = ResponseHandler(loop=loop)
        return StreamReader(protocol, limit=2 ** 16, loop=loop)

else:

    def stream_reader_factory(
        loop: Optional[asyncio.AbstractEventLoop] = None
    ) -> StreamReader:
        return StreamReader()
```